tutorbackup/templates/backup/build/backup/backup_services.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+
+import logging
+import os
+import shutil
+import sys
+import tarfile
+from subprocess import check_call
+
+import click
+from botocore.exceptions import ClientError
+
+
+ENV = os.environ
+
+DUMP_DIRECTORY = '/data'
+
+MYSQL_DUMPFILE = os.path.join(DUMP_DIRECTORY, 'mysql_dump.sql')
+MONGODB_DUMPDIR = os.path.join(DUMP_DIRECTORY, 'mongodb_dump')
+CADDY_DUMPDIR = os.path.join(DUMP_DIRECTORY, 'caddy')
+
+TARFILE = '/backup/backup.tar.xz'
+
+logger = logging.getLogger(__name__)
+
+PERFORM_CADDY_BACKUP = os.path.isdir('caddy')
```

Comment:
I don't think it's a good idea to make the backup depend on the existence of this directory. If we want to include Caddy data and it's not there, then we want this to blow up rather than succeed and silently hide the fact that the data is missing.

Reply:
Whether the Caddy data directory is mounted or not is [governed by the `ENABLE_WEB_PROXY` config parameter](https://github.com/hastexo/tutor-contrib-backup/blob/main/tutorbackup/patches/k8s-jobs#L58)
If it is not enabled, there is nothing to back up. I don't think that should cause the script to blow up. It's not an error state.
However, I agree that relying on the existence of the directory is not ideal as we are relying on an effect instead of the cause (ENABLE_WEB_PROXY=False). We can determine whether Caddy should be backed up or not outside and instruct the script to skip Caddy with a Click argument of some sort.

Reply:
Yes, that's kind of exactly what I suggested [here](https://github.com/hastexo/tutor-contrib-backup/pull/12#discussion_r843730044). :) Meaning, you can inject `{% if not ENABLE_WEB_PROXY %} --exclude caddy{% endif %}` into the `args` list in `k8s-jobs`.